File: src/main/java/org/usfirst/frc/team4099/robot/ControlBoard.java
package org.usfirst.frc.team4099.robot;

import org.usfirst.frc.team4099.lib.joystick.JoystickUtils;
import org.usfirst.frc.team4099.lib.joystick.LogitechF310Gamepad;

public class ControlBoard {
    private static ControlBoard sInstance = new ControlBoard();
    public static ControlBoard getInstance() {
        return sInstance;
    }
    private final LogitechF310Gamepad driver;
//    private final DualShock4Gamepad operator;

    private ControlBoard() {
        driver = new LogitechF310Gamepad(Constants.Joysticks.DRIVER_PORT);
//        driver = new DualShock4Gamepad(Constants.Joysticks.DRIVER_PORT);
//        operator = new DualShock4Gamepad(Constants.Joysticks.SHOTGUN_PORT);
    }

    public double getThrottle() {
        return -driver.getRightTriggerAxis() + driver.getLeftTriggerAxis();
    }

    public double getTurn() {
        return driver.getLeftXAxis();
    }

    public boolean getToggleSlowMode() {
        return driver.getAButton();
    }

    /**
     * Should the bot arcadeDrive in quick turn mode?
     * @return  true/false, depending on if the joystick is depressed
     */
    public boolean getQuickTurn() {
        return Math.abs(JoystickUtils.deadbandNoShape(getThrottle(), 0.02)) < 0.01;
        //return driver.getXButton();
    }

    public boolean getIntakeUp() {
        return driver.getDPadUp();
    }

    public boolean getIntakeDown() {
        return driver.getDPadDown();
    }

    public boolean getToggleIntake() {
        return driver.getAButton();
    }

    public boolean getClimber() {
        return driver.getYButton();
    }

    public boolean getToggleIntakeClosed() {
        return driver.getDPadRight();
    }

}

File: src/main/java/org/usfirst/frc/team4099/robot/ControlBoard.kt
package org.usfirst.frc.team4099.robot

import org.usfirst.frc.team4099.lib.joystick.Gamepad
import org.usfirst.frc.team4099.lib.joystick.JoystickUtils
import org.usfirst.frc.team4099.lib.joystick.LogitechF310Gamepad
import org.usfirst.frc.team4099.lib.joystick.XboxOneGamepad

class ControlBoard private constructor() {
    private val driver: Gamepad = LogitechF310Gamepad(Constants.Joysticks.DRIVER_PORT)
    private val operator: Gamepad = XboxOneGamepad(Constants.Joysticks.SHOTGUN_PORT)

    val throttle: Double
        get() = driver.rightTriggerAxis - driver.leftTriggerAxis

    val turn: Double
        get() = -driver.leftXAxis

    val switchToHighGear: Boolean
        get() = driver.rightShoulderButton

    val switchToLowGear: Boolean
        get() = driver.leftShoulderButton

    val openIntake: Boolean
        get() = operator.dPadLeft

    val closeIntake: Boolean
        get() = operator.dPadRight

    /**
     * Should the bot arcadeDrive in quick turn mode?
     * @return  true/false, depending on if the joystick is depressed
     */
    val quickTurn: Boolean
        get() = Math.abs(JoystickUtils.deadbandNoShape(throttle, 0.02)) < 0.01

    val reverseIntake: Boolean
        get() = operator.bButton

    val test: Boolean
        get() = operator.aButton

//    val elevatorTop: Boolean
//        get() = operator.dPadUp
//
//    val elevatorBottom: Boolean
//        get() = operator.dPadDown

//    val wristTop: Boolean
//        get() = driver.dPadUp
//
//    val wristBottom: Boolean
//        get() = driver.dPadDown

    val elevatorPower: Double
        get() = (operator.rightTriggerAxis - operator.leftTriggerAxis) * 1

    val wristPower: Double
        get() = operator.leftYAxis

    companion object {
        val instance = ControlBoard()
    }

}
